plugins/controlboard/include/yarp/dev/ControlBoardDriverCoupling.h

```diff
@@ -164,4 +164,27 @@ class CerHandCouplingHandler : public BaseCouplingHandler
     yarp::sig::Vector decoupleRefTrq (yarp::sig::Vector& trq_ref);
 };
 
+class HandMk3CouplingHandler : public BaseCouplingHandler
+{
+
+public:
+    HandMk3CouplingHandler (gazebo::physics::Model* model, yarp::sig::VectorOf<int> coupled_joints, std::vector<std::string> coupled_joint_names);
+
+public:
+    bool decouplePos (yarp::sig::Vector& current_pos);
+    bool decoupleVel (yarp::sig::Vector& current_vel);
+    bool decoupleAcc (yarp::sig::Vector& current_acc);
+    bool decoupleTrq (yarp::sig::Vector& current_trq);
+
+    yarp::sig::Vector decoupleRefPos (yarp::sig::Vector& pos_ref);
+    yarp::sig::Vector decoupleRefVel (yarp::sig::Vector& vel_ref);
+    yarp::sig::Vector decoupleRefTrq (yarp::sig::Vector& trq_ref);
+    
+protected:
+    double decouple (double q2, double lut[]);
+
+    double thumb_lut[4096];
```

Comment:
Instead of duplicating the value 4096 around, could be set the 4096 as a constexpr integer attribute and reuse it in the class? To be honest I am not sure if this works to set the size of a C-style array, but it should be easy to try.

Reply:
Done.

Reply:
It didn't work so I made the look up tables dynamically allocated.